source/index.js

```diff
@@ -156,14 +197,23 @@ module.exports = async (input = 'patch', options) => {
 		title: 'Pushing tags',
 		skip: async () => {
 			if (!(await git.hasUpstream())) {
-				return 'Upstream branch not found: not pushing.';
+				return 'Upstream branch not found; not pushing.';
+			}
+
+			if (!isPublished && runPublish) {
+				return 'Couldn\'t publish package to npm; not pushing.';
 			}
 		},
 		task: () => git.push()
 	});
 
 	tasks.add({
 		title: 'Creating release draft on GitHub',
+		skip: async () => {
+			if (!isPublished && runPublish) {
+				return 'Couldn\'t publish package to npm; not creating release draft.';
+			}
+		},
```

Comment:
It feels weird to add this here. It shouldn't even get this far. If the publish tasks fails, no other tasks should run.

Reply:
I added this to make sure that a release doesn't get created after the publish failed, but as you said, if publishing fails no other tasks should run.